### PR TITLE
chore: add streaming to bindings and use it

### PR DESCRIPTION
**Pull request by rb-determined-ai**
## Test Plan

Our logging has great CI already, so existing tests already provide sufficient coverage of refactored code.

## Commentary (optional)

After this, we will be able to add log following to the python sdk, which IMO is a big missing feature.

This change overlaps slightly with #4914.  It's unclear if that can land first since we have to track down some user behavior before landing it.

### CODE REVIEW

**Review by hamidzr — COMMENTED** [review]
i reviewed 754d6cfb88a9672cc7ab497a64bc99f5c7d529f7 and that lgtm. thanks! I think I saw a ticket asking for demos on how to get the trial logs in our Jira. they'd probably like this

File: harness/determined/common/api/bindings.py [hunk 1]
```diff
@@ -44,6 +45,19 @@ def __str__(self) -> str:
         return self.message
 
 
+class APIHttpStreamError(APIHttpError):
+    # APIHttpStreamError is used if an streaming API request fails mid-stream.
+    def __init__(self, operation_name: str, error: "runtimeStreamError") -> None:
+        self.operation_name = operation_name
+        self.error = error
+        self.message = (
+            f"Stream Error during {operation_name}: {error.message}"
```
[the remark this answers is not in the document]
Comment (rb-determined-ai) [reply harness/determined/common/api/bindings.py:54]: Hm, I think the error handling could be cleaned up for sure.

Right now, I think there's really just a missing error type... the APIHttpStreamError I added is specific to the in-band `runtimeStreamError`s that grpc can emit.  We probably need something for the in-band `runtimeError`s that grpc can emit, as well as the out-of-band errors that can occur (like connection errors, or if you tried talking to a non-grpc error or something).

I don't think our current error handling is very good.  So I am happy to make whatever change you think is best for this pr...?

[the remark this answers is not in the document]
Comment (hamidzr) [reply harness/determined/common/api/bindings.py:54]: I'd lean toward not addresing all of that work here. I see `error.message` as similar to getting the `Response.message` (or status text) in the basic error messages which is something we don't do so I thought it'd be nice to do the same in both cases nothing more than that for this pr.

**Review by stoksc — APPROVED** [review]
lgtm

**Pull request merged**